### ISSUE TITLE
assocPath inconsistent behavior

### ISSUE THREAD

**Issue by char0n** [ramda/ramda#2121]
Hi,

I detected inconsistent behavior on assocPath while debugging some production code. Let me demonstrate on example.

```javascript
R.assocPath(['test'], 3, null) -> Object({test: 3})
```

```javascript
R.assocPath(['test', 'test1'], 3, null) -> @Throws Uncaught TypeError: Cannot convert undefined or null to object
```

As you can see, the behavior change with size of the path. It seems that if path.length > 1, it always fails.

**Comment by buzzdecafe** [comment]
yeah, looks like the code handles empty list and lists with `length > 1`, but not `length = 1` case.  IMO the failure behavior is preferable; You pass a `null` where an `Object` is expected, then you have voided the warranty.
